src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -616,6 +616,8 @@ private async Task<ConnectBlocksResult> FullyValidateLockedAsync(ChainedHeader n
 
                         if (disconnectedBlock.ChainedHeader.Block == null)
                             disconnectedBlock.ChainedHeader.Block = disconnectedBlock.Block;
+
+                        this.chainedHeaderTree.SetAsUnconsumed(disconnectedBlock.ChainedHeader.Block);
```

Comment:
So when a block disconnects, it goes back to being considered unconsumed?
Sounds good to me.

Reply:
I don't think so, because I think that UnconsumedBlocks are block that are outside of the main chain, so even if you didn't load from store, it means you are still holding memory for a block not in the main chain (and I think that UnconsumedBlock is a name that doesn't fit what it really holds)

Reply:
No I mean because this memory is already calculated in the unconsumed counter.